lib/src/connection_fsm.dart

```diff
@@ -255,14 +255,13 @@ class _PostgreSQLConnectionStateBusy extends _PostgreSQLConnectionState {
     // print("(${query.statement}) -> $message");
 
     if (message is ReadyForQueryMessage) {
-      if (message.state == ReadyForQueryMessage.StateTransactionError) {
-        query.completeError(returningException);
-        return _PostgreSQLConnectionStateReadyInTransaction(
-            query.transaction as _TransactionProxy);
-      }
-
       if (returningException != null) {
-        query.completeError(returningException);
+        if (message.state == ReadyForQueryMessage.StateTransactionError) {
```

Comment:
How sure are you that `message.state == ReadyForQueryMessage.StateTransactionError` and `returningException == null` is never true at the same time? Maybe this should handle it as the original code?

Reply:
That was a bit of a mistake on my part, I nested ```message.state == ReadyForQueryMessage.StateTransactionError``` within ```message is ReadyForQueryMessage```, because the first statement did a type check that made the second statement safe. Nesting it inside the ```returningException != null``` if statement was overstepping.

Reply:
I think I may be misremembering my reasoning here.